src/lexer/mod.rs

```diff
@@ -245,6 +247,15 @@ impl Token {
                             ));
                         }
                     }
+                } else if is_generic_parameter(&ident_or_number) {
+                    if ident_or_number.len() < 2 {
+                        return Err(ctx.error(
+                            ErrorKind::NoOneLetterVariable,
+                            Span::new(ctx.filename_id, ctx.offset, 1),
+                        ));
+                    }
+
+                    TokenKind::Generic(ident_or_number)
```

Comment:
should be deleted imo, there's nothing special about a generic parameter that makes it stand out from an identifier

Reply:
deteled in another PR